javatests/arcs/core/allocator/AllocatorTestBase.kt

```diff
@@ -538,4 +555,90 @@ open class AllocatorTestBase {
             ArcState.Error
         )
     }
+
+    @Test
+    fun allocator_logArcStartAndStopToAnalytics() = runAllocatorTest {
+        val arcId = allocator.startArcForPlan(
+            "readWriteParticle",
+            PersonPlan
+        )
+
+        // Verify logs are recorded.
+        verify(mockLogger, times(1)).logAllocatorEvent(
+            eq(Analytics.Event.StartArc), eq("readWriteParticle"), any())
+        verify(mockLogger, times(1)).logArcHostEvent(
+            eq(Analytics.Event.StartArc), contains("arcs.core.host.TestingJvmProdHost"), any())
+        verify(mockLogger, times(1)).logArcHostEvent(
+            eq(Analytics.Event.StartArc), contains("arcs.core.allocator.AllocatorTestBase.ReadingHost"), any())
+        verify(mockLogger, times(1)).logArcHostEvent(
+            eq(Analytics.Event.StartArc), contains("arcs.core.allocator.AllocatorTestBase.WritingHost"), any())
+
+        allocator.stopArc(arcId)
+        // Verify logs are recorded.
+        verify(mockLogger, times(1)).logAllocatorEvent(
+            eq(Analytics.Event.StopArc), eq("readWriteParticle"), any())
+        verify(mockLogger, times(1)).logArcHostEvent(
+            eq(Analytics.Event.StopArc), contains("arcs.core.host.TestingJvmProdHost"), any())
+        verify(mockLogger, times(1)).logArcHostEvent(
+            eq(Analytics.Event.StopArc), contains("arcs.core.allocator.AllocatorTestBase.ReadingHost"), any())
+        verify(mockLogger, times(1)).logArcHostEvent(
+            eq(Analytics.Event.StopArc), contains("arcs.core.allocator.AllocatorTestBase.WritingHost"), any())
+
+        // Verify no more log is recorded.
+        verifyNoMoreInteractions(mockLogger)
+    }
+
+    @Test
+    fun allocator_logParticleNotFoundExceptionToAnalytics() = runAllocatorTest {
+        hostRegistry.unregisterHost(readingExternalHost)
```

Comment:
This won't work on Android since hosts are static and immutable based on AndroidManifest.xml. Instead, move this test to CoreAllocatorTest to make it Jvm/ExplicitHostRegistry only.

Reply:
You can make it cross platform by making AbstractArcHost.particleConstructors a protected field, and then in TestingHost add a method like unregister(ParticleIdentifier) and particleConstructors.remove(identifier),  then you can do readingExternalHost.unregister(FooParticle::class.toIdentifier())